### PR TITLE
Tabs: add suffix prop for the tab name

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
- Adds the feature highlights feature toggles in the config file
- Adds a `suffix` prop to the `Tab` component to be able to add a `React.FC` component in a tab header (after the tab label) and uses it in the `ShareModal` component

### CODE REVIEW

File: packages/grafana-ui/src/components/Tabs/Tab.tsx [hunk 1]
```diff
@@ -18,17 +18,19 @@ export interface TabProps extends HTMLProps<HTMLAnchorElement> {
...
-  ({ label, active, icon, onChangeTab, counter, className, href, ...otherProps }, ref) => {
+  ({ label, active, icon, onChangeTab, counter, suffix, className, href, ...otherProps }, ref) => {
     const theme = useTheme2();
     const tabsStyles = getTabStyles(theme);
     const content = () => (
       <>
         {icon && <Icon name={icon} />}
         {label}
         {typeof counter === 'number' && <Counter value={counter} />}
+        {suffix && <span>{suffix({})}</span>}
```
Comment (AgnesToulet) [inline packages/grafana-ui/src/components/Tabs/Tab.tsx:33]: Not sure at all about this syntax.

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Tabs/Tab.tsx:33]: I think it'd be possible just to do `{suffix}` here.

File: public/app/features/dashboard/components/ShareModal/types.ts [hunk 2]
```diff
@@ -10,5 +10,6 @@ export interface ShareModalTabProps {
 export interface ShareModalTabModel {
   label: string;
   value: string;
+  labelSuffix?: FC;
```
Comment (Clarity-89) [inline public/app/features/dashboard/components/ShareModal/types.ts:13]:
```suggestion
  labelSuffix?: JSX.Element;
```

**Review by Clarity-89 — COMMENTED** [review]
Left a few comments on how to better type the suffix. Also you need  to add milestone to a PR for the pipeline to be green.

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**